src/field.rs

```diff
@@ -190,6 +196,55 @@ pub trait FieldElement:
         }
         Ok(vec)
     }
+
+    /// Encode `input` as `bits`-bit vector of elements of `Self` if it's small enough
```

Comment:
I'm a bit wary of leaking these methods in the public API, but since @tgeoghegan suggested it feel free to ignore me :)
* encode_into_bitvector_representation
* decode_from_bitvector_representation
* valid_integer_try_from

Reply:
Not a bad point there, @cjpatton. The goal here was to put these new methods on all implementations of `FieldElement` without having them be floating functions in other modules. If we want to hide them from clients outside the crate, we could move them to some new trait `FieldElementExt` that is `pub(crate)`. I don't want to make Mxm do that though since this PR has been hanging out long enough as it is.